File: threat-detection-platform/backend/app/schemas/evidence.py
"""
Evidence Schemas
"""

from typing import Dict, List, Optional

from pydantic import BaseModel, Field


class EvidenceResponse(BaseModel):
    """Evidence metadata response."""

    id: str
    incident_id: str
    evidence_type: str  # snapshot, clip
    file_name: str
    file_size_bytes: int
    mime_type: str
    width: int | None = None
    height: int | None = None
    duration_seconds: float | None = None
    camera_id: str = ""
    captured_at: str
    expires_at: str | None = None
    is_expired: bool = False
    privacy_mode_applied: str = "off"
    faces_anonymized: int = 0

    model_config = {"from_attributes": True}


class EvidenceListResponse(BaseModel):
    """Evidence list for an incident."""

    incident_id: str
    items: List[EvidenceResponse]
    total: int

File: threat-detection-platform/frontend/src/types/evidence.ts
export interface EvidenceItem {
  id: string;
  incident_id: string;
  evidence_type: 'snapshot' | 'clip';
  file_name: string;
  file_size_bytes: number;
  mime_type: string;
  width: number | null;
  height: number | null;
  duration_seconds: number | null;
  camera_id: string;
  captured_at: string;
  expires_at: string | null;
  is_expired: boolean;
  privacy_mode_applied: 'off' | 'face_blur' | 'face_pixelate';
  faces_anonymized: number;
}

export interface EvidenceListResponse {
  incident_id: string;
  items: EvidenceItem[];
  total: number;
}
